### PR TITLE
docs: add ivy ssr opt-out

### LINKED ISSUE #15851: AppServerModuleNgFactory not exported for VE with version 9 approach
# 🐞 Bug report

### Command (mark with an `x`)

```
- [ ] new
- [x] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [ ] update
- [ ] lint
- [ ] xi18n
- [x] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc
```

### Is this a regression?
Yes

### Description
Currently in `ngtools` we rely on `isMainPath`, `getEntryModule` to run the `export_ngfactory` transformer, however in version 9 the main will be pointing to `server.ts` rather than `main.server.ts`. 

This causes the transformer to run on the wrong file in VE.

## 🔬 Minimal Reproduction

- npx i @angular/cli@9.0.0-next.9 new universal-test
- cd universal-test
- ng add @nguniversal/express-engine@9.0.0-next.5
- disable ivy
- in `server.ts` change `AppServerModule` to `AppServerModuleNgFactory` to use AOT as otherwise you will use the JIT version of the code.
-  npm run build:ssr

## 🔥 Exception or Error
<pre><code>
ERROR in server.ts:7:10 - error TS2305: Module '"C:/git/cli-repos/terser/src/main.server"' has no exported member 'AppServerModuleNgFactory'.

7 import { AppServerModuleNgFactory } from './src/main.server';
           ~~~~~~~~~~~~~~~~~~~~~~~~
</code></pre>


## 🌍 Your Environment
<pre><code>
Angular CLI: 9.0.0-next.9
Node: 12.10.0
OS: win32 x64
Angular: 9.0.0-next.10
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                           Version
-----------------------------------------------------------
@angular-devkit/architect         0.900.0-next.9
@angular-devkit/build-angular     0.900.0-next.9
@angular-devkit/build-optimizer   0.900.0-next.9
@angular-devkit/build-webpack     0.900.0-next.9
@angular-devkit/core              9.0.0-next.9
@angular-devkit/schematics        9.0.0-next.9
@angular/cli                      9.0.0-next.9
@angular/platform-server          9.0.0-next.11
@ngtools/webpack                  9.0.0-next.9
@nguniversal/common               9.0.0-next.5
@nguniversal/express-engine       9.0.0-next.5
@schematics/angular               9.0.0-next.9
@schematics/update                0.900.0-next.9
rxjs                              6.5.3
typescript                        3.5.3
webpack                           4.41.0
</code></pre>

**Anything else relevant?**
As a workaround users need to manually add the below export in `main.server.ts`

```ts
export { AppServerModuleNgFactory } from './app/app.server.module.ngfactory';
```

### PR DESCRIPTION
**Pull request by alan-agius4**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [x] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

fixes: https://github.com/angular/angular-cli/issues/15851

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: aio/content/guide/ivy.md [hunk 1]
```diff
@@ -70,17 +71,92 @@ The following example shows how to set the `enableIvy` option to `false` in orde
...
\ No newline at end of file
+<div class="alert is-important">
+
+If you disable Ivy, you might also want to reconsider whether to make AOT compilation the default for your application development, as described [above](#aot-and-ivy).
+
+To revert the compiler default, set the build option `aot: false` in the `angular.json` configuration file.
+
+</div>
+
+If your application uses [Angular Universal](guide/universal) to render Angular applications on the server. You also need to update the `server.ts` and provide the `AppServerModuleNgFactory` as the bootstrap module.
```
Comment (jbogarthyde) [inline aio/content/guide/ivy.md]: Is this what you need to do when you opt out of ivy?
How about a header:

### Using SSR without Ivy

If you opt out of Ivy and your application uses [Angular Universal](guide/universal) to render Angular applications on the server, you must also change the way the server performs bootstrapping.

File: aio/content/guide/ivy.md [hunk 2]
```diff
@@ -70,17 +71,92 @@ The following example shows how to set the `enableIvy` option to `false` in orde
...
+
+Follow these steps:
+* Import `AppServerModuleNgFactory` from the `app.server.module.ngfactory` virtual file.
+* Set the value of `bootstrap` option in the `ngExpressEngine` call to `AppServerModuleNgFactory`.
+
+The following example shows how the server bootstrapping should be changed to opt-out of ivy.
+
```
Comment (jbogarthyde) [inline aio/content/guide/ivy.md]: The following example shows how you modify the `server.ts` file to provide the `AppServerModuleNgFactory` as the bootstrap module.
* Import `AppServerModuleNgFactory` from the `app.server.module.ngfactory` virtual file.
* Set `bootstrap: AppServerModuleNgFactory` in the `ngExpressEngine` call.

**Review by jbogarthyde — CHANGES_REQUESTED** [review]
Edit text

File: aio/content/guide/ivy.md (hunk 1 above)
Comment (kapunahelewong) [inline aio/content/guide/ivy.md]: Little typo
```suggestion
If your application uses [Angular Universal](guide/universal) to render Angular applications on the server, you also need to update the `server.ts` and provide the `AppServerModuleNgFactory` as the bootstrap module.
```

**Comment by alan-agius4** [comment]
@kapunahelewong & @jbogarthyde PTAL. Thanks

**Review by kapunahelewong — APPROVED** [review]
LGTM, but deferring to @jbogarthyde as she has more specifics in mind. Thank you, @santoshyadav198613 :)

**Comment by alan-agius4** [comment]
Wrong user @kapunahelewong :P

**Review by jbogarthyde — APPROVED** [review]
Looks good - thanks!

**Comment by kapunahelewong** [comment]
> Wrong user @kapunahelewong :P

Oh!! 🙃 sorry, @alan-agius4! 🤦🏻‍♀️🎉

**Review by vikerman — APPROVED** [review]

**Pull request closed without merging**